File: 3066-minimum-operations-to-exceed-threshold-value-ii/3066-minimum-operations-to-exceed-threshold-value-ii.py
from sortedcontainers import *
class Solution(object):
    def minOperations(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        nums = SortedList(nums)
        a = 0
        while nums[0] < k:
            b = nums.pop(0)
            c = nums.pop(0)
            d = min(b, c) * 2 + max(b, c)
            nums.add(d)
            a += 1
        return a

File: 3066-minimum-operations-to-exceed-threshold-value-ii/3066-minimum-operations-to-exceed-threshold-value-ii.java
import java.util.PriorityQueue;

class Solution {
    public int minOperations(int[] nums, int k) {
        PriorityQueue<Long> pq = new PriorityQueue<>();
        for (int num : nums) {
            pq.add((long) num);
        }
        long cnt = 0;
        while (pq.peek() < k) {
            long first = pq.poll();
            long second = pq.poll();
            long num = 2 * Math.min(first, second) + Math.max(first, second);
            pq.add(num);
            cnt++;
        }
        return (int) cnt;
    }
}